benchmark/benchmark.rb

```diff
@@ -9,15 +9,58 @@
 end
 file = File.new(file)
 
-count = count2 = count3 = count4 = 0
-Benchmark.bm do |x|
-  x.report('for_tag') { Saxerator.parser(file).for_tag(:artist).each { count = count + 1 } }
-  x.report('at_depth') { Saxerator.parser(file).at_depth(2).each { count2 = count2 + 1 } }
-  x.report('within') { Saxerator.parser(file).within(:artists).each { count3 = count3 + 1 } }
-  x.report('composite') { Saxerator.parser(file).for_tag(:name).within(:artist).at_depth(3).each { count4 = count4 + 1} }
+ADAPTERS = [:nokogiri, :ox]
+
+class SaxeratorBenchmark
+
+  def initialize(file)
+    @file = file
+  end
+
+  def with_adapter(adapter)
+    puts "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"
+    puts
+    puts "Benchmark with :#{adapter} parser"
+    puts
+    puts "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@"
+    puts
+
+
+    count = count2 = count3 = count4 = 0
+
+    Benchmark.bm do |x|
+      x.report('for_tag') do
+        Saxerator.parser(@file) { |confing| confing.adapter = adapter}
+          .for_tag(:artist).each { count = count + 1 }
+      end
+
+      x.report('at_depth') do
+        Saxerator.parser(@file) { |confing| confing.adapter = adapter}
+          .at_depth(2).each { count2 = count2 + 1 }
+      end
+
+      x.report('within') do 
+        Saxerator.parser(@file){ |confing| confing.adapter = adapter}
+          .within(:artists).each { count3 = count3 + 1 }
+      end
+
+      x.report('composite') do
+        Saxerator.parser(@file){ |confing| confing.adapter = adapter}
+          .for_tag(:name).within(:artist).at_depth(3).each { count4 = count4 + 1}
+      end
+    end
+
+    puts
+    puts "##########################################################"
+    puts
+    puts "for_tag:   #{count} artist elements parsed"
+    puts "at_depth:  #{count2} elements parsed"
+    puts "within:    #{count3} artists children parsed"
+    puts "composite: #{count4} names within artist nested 3 tags deep parsed"
+    puts
+  end
 end
 
-puts "for_tag:   #{count} artist elements parsed"
-puts "at_depth:  #{count2} elements parsed"
-puts "within:    #{count3} artists children parsed"
-puts "composite: #{count4} names within artist nested 3 tags deep parsed"
+saxerator_benchmark = SaxeratorBenchmark.new(file)
```

Comment:
This is awesome that you updated the benchmarks!

```
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Benchmark with :nokogiri parser

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

       user     system      total        real
for_tag  5.800000   0.020000   5.820000 (  5.823733)
at_depth  4.590000   0.020000   4.610000 (  4.605400)
within  5.640000   0.010000   5.650000 (  5.657823)
composite  4.310000   0.020000   4.330000 (  4.328335)

##########################################################

for_tag:   100000 artist elements parsed
at_depth:  400000 elements parsed
within:    100000 artists children parsed
composite: 0 names within artist nested 3 tags deep parsed

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Benchmark with :ox parser

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

       user     system      total        real
for_tag  3.710000   0.010000   3.720000 (  3.723125)
at_depth  2.710000   0.010000   2.720000 (  2.723924)
within  3.630000   0.010000   3.640000 (  3.644329)
composite  2.130000   0.010000   2.140000 (  2.139019)

##########################################################

for_tag:   100000 artist elements parsed
at_depth:  400000 elements parsed
within:    100000 artists children parsed
composite: 0 names within artist nested 3 tags deep parsed
```

Seems like the composite one is slightly broken, but that's unrelated to this changeset. Ox is faster 👍